implementations/elixir/ockam/ockam/lib/ockam/examples/forwarding/forwarding.ex

```diff
@@ -65,11 +65,14 @@ defmodule Ockam.Examples.Forwarding do
 
     Ockam.Node.register_address("example_initiator")
 
-    Ockam.Router.route(%{
-      onward_route: forwarder_route,
-      return_route: ["example_initiator"],
-      payload: "Hello forward"
-    })
+    Ockam.Worker.route(
```

Comment:
this doesn't make sense,  it is _not_  a worker making this call, it doesn't have any state to pass around (as you noted passing it a nil) .   The relevant part of the issue is
> and changing all usages of Ockam.Router.route/1 from context of workers to Ockam.Worker.route/2

In general,   there shouldn't be any case in which Ockam.Worker.route/2 is called with _nil_ as state.

Reply:
Good point, though I inquired [about this ](https://github.com/build-trust/ockam/issues/3654#issuecomment-1441232891) and was under the impression that *all* instances would use Worker, not Route, but I can now understand the confusion and will revert where it's not necessary.

Reply:
@polvorin For clarification, in `examples/messaging/shuffle.ex`, the function originally was: 
```elixir
def forward_message(message, _state)
  ...
  Ockam.Router.route(Message.forward(message))
```
My PR updates it to make state available (because it was ignored), and then pass it into the Worker. Is this the desired pattern?